src/engine/pieces/piece.ts

```diff
@@ -13,8 +17,35 @@ export default class Piece {
         throw new Error('This method must be implemented, and return a list of available moves');
     }
 
+    public addAvailableMoveAndContinue(availableMoves: Array<Square>, newRow : number, newCol : number, currentPosition : Square, board : Board){
+        let newPos = new Square(newRow, newCol);
+        let newPosInBound = newPos.inBoundsCheck();
+        if(!currentPosition.equals(newPos) && newPosInBound){
```

Comment:
you could simplify this with early returns e.g.
```
    if (!newPos.inBoundsCheck()) {
        return false;
    }

    if (currentPosition.equals(newPos)) {
        return false;
    }
```

then no need for the big if statement and it'd hopefully be a bit clearer

Reply:
We are unsure of how we could use early returns :(

Reply:
can discuss in person